components/views/card-view.js

```diff
@@ -0,0 +1,116 @@
+                    this.render();
+                },
+
+                switchOnConfiguration: function (square_result, list_result) {
+                    return this.callIfFunction(this.configuration) === 'square_card' ?
+                        square_result : list_result;
+                },
+
+                callIfFunction: function (value) {
+                    if ($.isFunction(value)) {
+                        return value.call(this);
+                    } else {
+                        return value;
+                    }
+                },
+
+                className: function () {
+                    var result = 'card ' +
+                                 this.switchOnConfiguration('square-card', 'list-card') + ' ' +
+                                 this.callIfFunction(this.cardClass);
+                    if (this.callIfFunction(this.pennant)) {
+                        result += ' has-pennant';
+                    }
+                    return result;
+                },
+
+                render: function () {
+                    var maxLength = 72,
+                        description = this.callIfFunction(this.description),
+                        configuration = this.callIfFunction(this.configuration),
+                        template = _.template(
+                            configuration === 'square-card' ? squareCardViewTemplate : basicCardViewTemplate
+                        );
+                    if (description.length > maxLength) {
+                        description = description.substring(0, maxLength).trim() + '...';
+                    }
+                    this.$el.html(template({
+                        pennant: this.callIfFunction(this.pennant),
+                        title: this.callIfFunction(this.title),
+                        description: description,
+                        action_class: this.callIfFunction(this.actionClass),
+                        action_url: this.callIfFunction(this.actionUrl),
+                        action_content: this.callIfFunction(this.actionContent),
+                        configuration: configuration,
+                        srInfo: this.srInfo
+                    }));
+                    var detailsEl = this.$el.find('.card-meta');
+                    _.each(this.callIfFunction(this.details), function (detail) {
+                        // Call setElement to rebind event handlers
+                        detail.setElement(detail.el).render();
+                        detail.$el.addClass('meta-detail');
+                        detailsEl.append(detail.el);
+                    });
+                    return this;
+                },
+
+                action: function () { },
+                cardClass: '',
+                pennant: '',
+                title: '',
+                description: '',
+                details: [],
+                actionClass: '',
+                actionUrl: '',
+                actionContent: ''
```

Comment:
As a convention I think it is easier to read if the variables are declared before any of the functions (i.e. lines 104-111 being above the constructor). Thoughts @andy-armstrong @dsjen ?